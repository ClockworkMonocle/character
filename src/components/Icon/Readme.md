Power of the custom Hippo icons, plus all of font-awesome

```js
const hippoIcons = [
  'graph',
  'crown',
  'shield',
  'talk-chat',
  'talk-chat-group',
  'beer',
  'repo',
  'dollar',
  'group',
  'link',
  'link-broken',
  'chart',
  'attack',
  'equipment',
  'features',
  'anchor',
  'branching',
  'angle-double-right',
  'cog',
];

<div>
  {hippoIcons.map(ico => (
    <div style={{ display: 'flex', alignItems: 'center', marginBottom: '1em' }}>
      <Icon name={ico} key={ico} size={32} />
      <span style={{ marginLeft: '1em' }}>{ico}</span>
    </div>
  ))}
</div>
```

Sizes: 16, 32, 48, 64 pixels

```js
<div>
  <Icon name="user-secret" />
  <Icon size={32} name="user-secret" />
  <Icon size={48} name="user-secret" />
  <Icon size={64} name="user-secret" />
</div>
```

Color however you want:

```js
<Icon style={{ color: 'blue' }} name="arrow-up" size={64}/>
```
