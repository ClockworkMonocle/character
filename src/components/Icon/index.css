
.icon__16 {
  font-size: 16px;
}
.icon__32 {
  font-size: 32px;
}
.icon__48 {
  font-size: 48px;
}
.icon__64 {
  font-size: 64px;
}
