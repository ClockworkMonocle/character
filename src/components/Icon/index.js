import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import './index.css';

const hippoIcons = [
  'graph',
  'crown',
  'shield',
  'talk-chat',
  'talk-chat-group',
  'beer',
  'repo',
  'dollar',
  'group',
  'link',
  'link-broken',
  'chart',
  'attack',
  'equipment',
  'features',
  'anchor',
  'branching',
  'angle-double-right',
  'cog',
];

export default function Icon({ name, size, className, ...props }) {
  const iconName = hippoIcons.includes(name)
    ? `icon-${name}`
    : `fa fa-${name}`;

  const iconSize = `icon__${size}`;

  return (
    <span className={cn(iconName, iconSize, className)} {...props} />
  );
}

Icon.propTypes = {
  name: PropTypes.string,
  size: PropTypes.oneOf([16, 32, 48, 64]),
};

Icon.defaultProps = {
  size: 16,
};
