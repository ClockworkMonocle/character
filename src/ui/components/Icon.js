'use strict';

var React = require('react/addons');

module.exports = React.createClass({
  displayName : "Icon",

  render : function() {
    return (<span className={this.props.icon}></span>);
  }
})