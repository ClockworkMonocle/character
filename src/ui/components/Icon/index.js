'use strict';

var React = require('react');

module.exports = React.createClass({
  displayName : "Icon",

  render : function() {
    var { icon, className, ...others } = this.props;

    return (
      <span
        {...others}
        className={`icon ${icon} ${className || ''}`}
      ></span>
    );
  }
})