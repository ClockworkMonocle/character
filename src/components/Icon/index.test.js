import React from 'react';
import { shallow } from 'enzyme';
import Icon from './index';

test('<Icon /> renders hippo icon', () => {
  const mount = shallow(<Icon name="beer" />);
  expect(mount.find('.icon-beer').length).toEqual(1);
  expect(mount).toMatchSnapshot();
});

test('<Icon /> renders font awesome icon', () => {
  const mount = shallow(<Icon name="arrow-up" />);
  expect(mount.find('.fa.fa-arrow-up').length).toEqual(1);
  expect(mount).toMatchSnapshot();
});
