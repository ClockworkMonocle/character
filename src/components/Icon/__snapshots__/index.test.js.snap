// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders font awesome icon 1`] = `
<span
  className="fa fa-arrow-up icon__16"
/>
`;

exports[`<Icon /> renders hippo icon 1`] = `
<span
  className="icon-beer icon__16"
/>
`;
